base_layer/core/src/base_node/state_machine_service/states/horizon_state_sync/state_sync.rs

```diff
@@ -43,7 +43,7 @@ use tari_crypto::tari_utilities::Hashable;
 
 const LOG_TARGET: &str = "c::bn::state_machine_service::states::horizon_state_sync";
 
-pub struct HorizonStateSynchronization<'a, B> {
+pub struct HorizonStateSynchronization<'a, B: BlockchainBackend> {
```

Comment:
Trait bounds should (almost) never be needed on struct/enum definitions and are [generally undesirable](https://rust-lang.github.io/api-guidelines/future-proofing.html#data-structures-do-not-duplicate-derived-trait-bounds-c-struct-bounds). BlockchainBackend is added here because of the unnecessary trait bound on the `CandidateBlockBodyValidation` trait. I suggest removing the trait bound from `CandidateBlockBodyValidation`, `SyncValidators`, `BaseNodeStateMachine` and here.